infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProject.java

```diff
@@ -37,20 +37,29 @@
   @VisibleForTesting
   OpenShiftProject(
       String workspaceId,
+      String name,
       KubernetesPods pods,
       KubernetesServices services,
       OpenShiftRoutes routes,
       KubernetesPersistentVolumeClaims pvcs,
       KubernetesIngresses ingresses) {
-    super(workspaceId, pods, services, pvcs, ingresses);
+    super(workspaceId, name, pods, services, pvcs, ingresses);
     this.routes = routes;
   }
 
   public OpenShiftProject(OpenShiftClientFactory clientFactory, String name, String workspaceId)
       throws InfrastructureException {
+    this(clientFactory, name, workspaceId, true);
+  }
+
+  public OpenShiftProject(
+      OpenShiftClientFactory clientFactory, String name, String workspaceId, boolean doPrepare)
```

Comment:
Can you add a description to the `doPrepare` parameter and constructors?

Reply:
Replaced `doPrepare` parameter with package private method that should be invoked by project factory.